lib/ansible/executor/task_executor.py

```diff
@@ -617,6 +617,14 @@ def _execute(self, variables=None):
             if self._task.async_val > 0:
                 if self._task.poll > 0 and not result.get('skipped') and not result.get('failed'):
                     result = self._poll_async_result(result=result, templar=templar, task_vars=vars_copy)
+                    if result.get('failed'):
+                        self._final_q.send_callback(
+                            'v2_runner_on_async_failed',
+                            TaskResult(self._host.name, self._task, result, task_fields=self._task.dump_attrs()))
```

Comment:
Not sure if we should copy the comment from async poll that indicates that `self._task` should be the full task because the strategy doesn't know about the task, since the TE created it.  Might alleviate some confusion later.

Reply:
Good point. Latest commit adds the comments.